pkg/controller/dnszone/gcpactuator.go

```diff
@@ -96,7 +101,8 @@ func (a *GCPActuator) Delete() error {
 
 	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone).WithField("zoneName", a.managedZone.Name)
```

Comment:
The `Delete` function verifies that `a.managedZone` is not nil but does not use `a.managedZone` for anything other than the logger fields. The `Delete` function **does** use `a.dnsZone.Status.GCP.ZoneName`, but never verifies that it is not nil.
```suggestion
	logger := a.logger.WithField("zone", a.dnsZone.Spec.Zone).WithField("zoneName", *a.dnsZone.Status.GCP.ZoneName)
```

Reply:
Sorry, `a.managedZone` **is** used. But it should be consistent in the `Delete` function. Use either `a.managedZone.Name` or `a.dnsZone.Status.GCP.ZoneName`. I suggest that the `managedZone` field should be removed entirely from the actuator.

Reply:
I think the primary utility is that it holds the list of nameservers to satisfy the actuator's GetNameServers() call. We could store it as another struct field after creating/fetching the DNSZone though.

Reply:
OK. That makes sense. I am uneasy about there being two different places from which information is obtained in the actuators--either from the `managedZone` in the actuator or from the status of the `DNSZone`. That is going to make it harder to maintain in the future because we are going to need to make sure that both are kept in sync. But I can live with it for now.